src/main/java/com/github/pedrovgs/androidwifiadb/adb/ADB.java

```diff
@@ -69,11 +72,40 @@ private String getDeviceIp(Device device) {
   }
 
   private boolean connectDevice(String deviceIp) {
-    String enableTCPCommand = getCommand("tcpip 5555");
-    commandLine.executeCommand(enableTCPCommand);
+    enableTCPCommand();
     String connectDeviceCommand = getCommand("connect " + deviceIp);
-    String connectOutput = commandLine.executeCommand(connectDeviceCommand);
-    return connectOutput.contains("connected");
+    return commandLine.executeCommand(connectDeviceCommand).contains("connected");
+  }
+
+  /**
+   * Disconnect connected device by IP.
+   * If device is disconnected with success, we should get an empty command result.
+   * @param deviceIp
+   * @return
+   */
+  private boolean disconnectDevice(String deviceIp) {
+    enableTCPCommand();
+    String connectDeviceCommand = getCommand("disconnect " + deviceIp);
+    return commandLine.executeCommand(connectDeviceCommand).isEmpty();
+  }
+
+  public List<Device> disconnectDevices(List<Device> devices) {
+    for (Device device : devices) {
+      boolean disconnected = disconnectDevice(device.getIp());
+      device.setConnected(disconnected);
+    }
+    return devices;
+  }
```

Comment:
Well, it's not 100% true, please check 498615ee0f0250a948135cda84cfce1e1abc68c1 commit for example. 
I'll update all the classes applying this rule :)